### PR TITLE
User can now select transaction Isolation for CTR

### LINKED ISSUE #4680: ctr error with transaction levels
I am using spring cloud dataflow server (2.8.1) , it is deployed on openshift (kubernetes deployer). 

we have set below properties to connect to oracle database.
spring.datasource.url: 
spring.datasource.username: 
spring.datasource.password: 
spring.datasource.driverClassName:

we are able to successfully launch the server and  able to successfully launch the task as well.
we have set the below properties to configure spring-cloud-composed-task-runner

spring.cloud.dataflow.task.composedtaskrunner.uri=<this point to image of runner uploaded on openshift>
dataflow-server-uri=<this is spring cloud data flow server url>

image of composedtask runner is created using below docker file

content of docker file
-----------------------------------------------------
FROM <docker registry>/fabric-managed-images/java-open-jdk:8-compliant
ENV TARGET_DIR="/deployments"           

RUN mkdir /${TARGET_DIR} \
    && chgrp -R 0 /${TARGET_DIR} \
    && chmod -R g+rwx /${TARGET_DIR}

RUN mkdir -p /${TARGET_DIR}/config/

COPY ./jars/spring-cloud-dataflow-composed-task-runner-2.8.1.jar /${TARGET_DIR}/
COPY ./jars/ojdbc6.jar /${TARGET_DIR}/
COPY ./entry.sh /${TARGET_DIR}/
COPY ./application.yaml /${TARGET_DIR}/config/

RUN ls -lrt ${TARGET_DIR}

RUN chmod -R 755 /${TARGET_DIR}/spring-cloud-dataflow-composed-task-runner-2.8.1.jar
RUN chmod -R 755 /${TARGET_DIR}/entry.sh
RUN chmod -R 755 /${TARGET_DIR}/config/application.yaml

ENTRYPOINT ["/deployments/entry.sh"]
CMD ["--help"]
--------------------------------------------------

content of entry.sh
-------------------------------------

#!/bin/bash

set -o errexit
set -o nounset
set -o pipefail

__run_cmd="java"
__run_flags=("-cp" "/deployments/spring-cloud-dataflow-composed-task-runner-2.8.1.jar" "-Dloader.path=/deployments/ojdbc6.jar" "-Dspring.config.location=/deployments/config/" "org.springframework.boot.loader.PropertiesLauncher" "$@")

echo  "command is:" "$__run_cmd" "${__run_flags[@]}"
exec "$__run_cmd" "${__run_flags[@]}"
--------------------------

Now when I launch the composed task in spring-cloud-dataflow server. I get  sql exception "READ_COMMITTED and SERIALIZABLE are the only valid transaction levels". below is complete log

```
command is: java -cp /deployments/spring-cloud-dataflow-composed-task-runner-2.8.1.jar -Dloader.path=/deployments/ojdbc6.jar -Dspring.config.location=/deployments/config/ org.springframework.boot.loader.PropertiesLauncher --spring.datasource.driverClassName=oracle.jdbc.OracleDriver --management.metrics.tags.application=${spring.cloud.task.name:unknown}-${spring.cloud.task.executionid:unknown} --dataflowServerUri=http://scdf-server-dk0679-i.uki1f.paas.intranet.db.com:80  --platform-name=default --spring.cloud.task.name=hybrid-pipeline-01 --graph=<hybrid-pipeline-01-log-hello-1 || hybrid-pipeline-01-py-test-08> --spring.datasource.password=test1 --management.metrics.tags.service=task-application --spring.datasource.username=SDF_OWNER --spring.datasource.url=jdbc:oracle:thin:@//frorr2d.de.db.com:1825/frorr2d_app.de.db.com --composed-task-app-properties.base64_ZGVwbG95ZXIuKi5jb25maWctbWFwLXJlZnM=hikary-cp --composed-task-app-properties.base64_ZGVwbG95ZXIuKi5rdWJlcm5ldGVzLmNvbmZpZy1tYXAtcmVmcw=hikary-cp --spring.cloud.task.executionid=125 --spring.cloud.data.flow.taskappname=composed-task-runner

  _____                                         _   _______        _
 / ____|                                       | | |__   __|      | |
| |     ___  _ __ ___  _ __   ___  ___  ___  __| |    | | __ _ ___| | __
| |    / _ \| '_ ` _ \| '_ \ / _ \/ __|/ _ \/ _` |    | |/ _` / __| |/ /
| |___| (_) | | | | | | |_) | (_) \__ \  __/ (_| |    | | (_| \__ \   <
 \_____\___/|_| |_| |_| .__/ \___/|___/\___|\__,_|    |_|\__,_|___/_|\_\
                      | |
                      |_|
 _____
|  __ \
| |__) |   _ _ __  _ __   ___ _ __
|  _  / | | | '_ \| '_ \ / _ \ '__|
| | \ \ |_| | | | | | | |  __/ |
|_|  \_\__,_|_| |_|_| |_|\___|_|

2021-09-13 12:20:56.173  INFO 1 --- [           main] o.s.c.d.c.ComposedTaskRunner             : Starting ComposedTaskRunner v2.8.1 using Java 1.8.0_301 on hybrid-pipeline-01-ny3nrj83oo with PID 1 (/deployments/spring-cloud-dataflow-composed-task-runner-2.8.1.jar started by ? in /)
2021-09-13 12:20:56.181  INFO 1 --- [           main] o.s.c.d.c.ComposedTaskRunner             : No active profile set, falling back to default profiles: default
2021-09-13 12:20:57.902  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Multiple Spring Data modules found, entering strict repository configuration mode!
2021-09-13 12:20:57.905  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2021-09-13 12:20:57.950  INFO 1 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 20 ms. Found 0 JPA repository interfaces.
2021-09-13 12:20:58.382  INFO 1 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.cloud.task.batch.configuration.TaskBatchAutoConfiguration' of type [org.springframework.cloud.task.batch.configuration.TaskBatchAutoConfiguration$$EnhancerBySpringCGLIB$$4bb95862] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-09-13 12:20:59.097  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2021-09-13 12:20:59.775  INFO 1 --- [           main] com.zaxxer.hikari.pool.PoolBase          : HikariPool-1 - Driver does not support get/set network timeout for connections. (oracle.jdbc.driver.T4CConnection.getNetworkTimeout()I)
2021-09-13 12:20:59.804  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2021-09-13 12:20:59.936  INFO 1 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: default]
2021-09-13 12:21:00.103  INFO 1 --- [           main] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 5.4.32.Final
2021-09-13 12:21:00.486  INFO 1 --- [           main] o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
2021-09-13 12:21:00.858  INFO 1 --- [           main] org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.Oracle12cDialect
2021-09-13 12:21:01.594  INFO 1 --- [           main] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
2021-09-13 12:21:01.617  INFO 1 --- [           main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
2021-09-13 12:21:01.965  INFO 1 --- [           main] o.s.b.c.r.s.JobRepositoryFactoryBean     : No database type set, using meta data indicating: ORACLE
2021-09-13 12:21:02.003  INFO 1 --- [           main] o.s.b.c.r.s.JobRepositoryFactoryBean     : No database type set, using meta data indicating: ORACLE
2021-09-13 12:21:02.004  INFO 1 --- [           main] o.s.b.c.l.support.SimpleJobLauncher      : No TaskExecutor has been set, defaulting to synchronous executor.
2021-09-13 12:21:03.895  INFO 1 --- [           main] o.s.c.d.c.ComposedTaskRunner             : Started ComposedTaskRunner in 9.282 seconds (JVM running for 10.586)

2021-09-13 12:21:03.898  INFO 1 --- [           main] o.s.b.a.b.JobLauncherApplicationRunner   : Running default command line with: []
2021-09-13 12:21:03.898  INFO 1 --- [           main] o.s.b.c.r.s.JobRepositoryFactoryBean     : No database type set, using meta data indicating: ORACLE
2021-09-13 12:21:04.199  INFO 1 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-09-13 12:21:04.251 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application run failed

java.lang.IllegalStateException: Failed to execute ApplicationRunner
    at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:802) [spring-boot-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:789) [spring-boot-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:346) [spring-boot-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1329) [spring-boot-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1318) [spring-boot-2.4.7.jar!/:2.4.7]
    at org.springframework.cloud.dataflow.composedtaskrunner.ComposedTaskRunner.main(ComposedTaskRunner.java:31) [classes!/:2.8.1]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_301]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_301]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_301]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_301]
    at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) [spring-cloud-dataflow-composed-task-runner-2.8.1.jar:2.8.1]
    at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) [spring-cloud-dataflow-composed-task-runner-2.8.1.jar:2.8.1]
    at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) [spring-cloud-dataflow-composed-task-runner-2.8.1.jar:2.8.1]
    at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:467) [spring-cloud-dataflow-composed-task-runner-2.8.1.jar:2.8.1]
Caused by: org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLException: READ_COMMITTED and SERIALIZABLE are the only valid transaction levels
    at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:309) ~[spring-jdbc-5.3.8.jar!/:5.3.8]
    at org.springframework.transaction.support.AbstractPlatformTransactionManager.startTransaction(AbstractPlatformTransactionManager.java:400) ~[spring-tx-5.3.8.jar!/:5.3.8]
    at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373) ~[spring-tx-5.3.8.jar!/:5.3.8]
    at org.springframework.transaction.interceptor.TransactionAspectSupport.createTransactionIfNecessary(TransactionAspectSupport.java:595) ~[spring-tx-5.3.8.jar!/:5.3.8]
    at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:382) ~[spring-tx-5.3.8.jar!/:5.3.8]
    at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-5.3.8.jar!/:5.3.8]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at com.sun.proxy.$Proxy60.getLastJobExecution(Unknown Source) ~[na:na]
    at org.springframework.batch.core.launch.support.SimpleJobLauncher.run(SimpleJobLauncher.java:104) ~[spring-batch-core-4.3.3.jar!/:4.3.3]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_301]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_301]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_301]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_301]
    at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at org.springframework.batch.core.configuration.annotation.SimpleBatchConfiguration$PassthruAdvice.invoke(SimpleBatchConfiguration.java:128) ~[spring-batch-core-4.3.3.jar!/:4.3.3]
    at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215) ~[spring-aop-5.3.8.jar!/:5.3.8]
    at com.sun.proxy.$Proxy63.run(Unknown Source) ~[na:na]
    at org.springframework.boot.autoconfigure.batch.JobLauncherApplicationRunner.execute(JobLauncherApplicationRunner.java:199) ~[spring-boot-autoconfigure-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.autoconfigure.batch.JobLauncherApplicationRunner.executeLocalJobs(JobLauncherApplicationRunner.java:173) ~[spring-boot-autoconfigure-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.autoconfigure.batch.JobLauncherApplicationRunner.launchJobFromProperties(JobLauncherApplicationRunner.java:160) ~[spring-boot-autoconfigure-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.autoconfigure.batch.JobLauncherApplicationRunner.run(JobLauncherApplicationRunner.java:155) ~[spring-boot-autoconfigure-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.autoconfigure.batch.JobLauncherApplicationRunner.run(JobLauncherApplicationRunner.java:150) ~[spring-boot-autoconfigure-2.4.7.jar!/:2.4.7]
    at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:799) [spring-boot-2.4.7.jar!/:2.4.7]
    ... 13 common frames omitted
Caused by: java.sql.SQLException: READ_COMMITTED and SERIALIZABLE are the only valid transaction levels
    at oracle.jdbc.driver.PhysicalConnection.setTransactionIsolation(PhysicalConnection.java:4638) ~[ojdbc6.jar:11.2.0.4.0]
    at com.zaxxer.hikari.pool.ProxyConnection.setTransactionIsolation(ProxyConnection.java:433) ~[HikariCP-3.4.5.jar!/:na]
    at com.zaxxer.hikari.pool.HikariProxyConnection.setTransactionIsolation(HikariProxyConnection.java) ~[HikariCP-3.4.5.jar!/:na]
    at org.springframework.jdbc.datasource.DataSourceUtils.prepareConnectionForTransaction(DataSourceUtils.java:214) ~[spring-jdbc-5.3.8.jar!/:5.3.8]
    at org.springframework.jdbc.datasource.DataSourceTransactionManager.doBegin(DataSourceTransactionManager.java:275) ~[spring-jdbc-5.3.8.jar!/:5.3.8]
```

Please help me to resolve this issue. I got stuck on this part from past 2 days

Any help is highly appreciated and I would  be glad to take any feedback for image creation part as well

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #4680

### CODE REVIEW

File: spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/properties/ComposedTaskProperties.java [hunk 1]
```diff
@@ -191,6 +191,12 @@
 	 */
 	private boolean skipTlsCertificateVerification = false;
 
+	/**
+	 * Establish the transaction isolation level for the Composed Task Runner.
+	 * Default is ISOLATION_REPEATABLE_READ.
+	 */
+	private String transactionIsolationLevel = "ISOLATION_REPEATABLE_READ";
```
Comment (jvalkeal) [inline spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/properties/ComposedTaskProperties.java:198]: Was going to nitpick that this is a string but looks like factory in batch is also using string. That was strange as in spring transaction classes these are enums. I would be nice to give a set of valid options but not sure how feasible that is.

Comment (jvalkeal) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/properties/ComposedTaskProperties.java:198]: It's worth to check if boot has any similar configuration options and if so then use same type.

Comment (cppwfs) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/properties/ComposedTaskProperties.java:198]: I spelunked boot, but didn't find anything that mapped a string version of transaction isolation level to an enumeration.

Comment (jvalkeal) [reply spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/properties/ComposedTaskProperties.java:198]: I found https://github.com/spring-projects/spring-boot/blob/3e21ae12dc2653ab7ce7d561c29bc018a57a4103/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/batch/JpaBatchConfigurer.java#L58 and that is just string. We can then use same String type even if enum would be better. At least it's same in both places.

**Comment by jvalkeal** [comment]
lgtm, but can you fix conflicts as other PR changed that class.

**Comment by cppwfs** [comment]
Rebased and resolved conflicts.

**Pull request merged**